addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1663,7 +1663,7 @@ var AbstractFieldBinary = AbstractField.extend({
         this._super.apply(this, arguments);
         this.fields = record.fields;
         this.useFileAPI = !!window.FileReader;
-        this.max_upload_size = 25 * 1024 * 1024; // 25Mo
+        this.max_upload_size = odoo.session_info.max_file_size;
```

Comment:
I'd rather keep a default here, in case the new JS code runs before the Python code gets reloaded (safer for a stable version).

```suggestion
        this.max_upload_size = odoo.session_info.max_file_size || 128 * 1024 * 1024;
```